server/Infrastructure/Repository/TeacherRepository.cs

```diff
@@ -22,6 +22,11 @@ public Teacher InsertTeacher(Teacher teacher)
             return entity.Entity;
         }
 
+        public Teacher GetTeacherById(int id)
+        {
+            return context.Teachers.AsNoTracking().FirstOrDefault(t => t.Id == id);
```

Comment:
this changes are not related to the task, I guess

Reply:
It was made during the process and will be useful soon, there is no reason to delete it and rewrite it again